tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -545,6 +531,10 @@ unique_ptr<DuckDBPyRelation> DuckDBPyConnection::ReadJSON(const string &name, co
 	return make_unique<DuckDBPyRelation>(std::move(read_json_relation));
 }
 
+PathLike DuckDBPyConnection::GetPathLike(const py::object &object) {
```

Comment:
I'm not at all familiar with the internals of duckdb, but I might consider leaving this out of the DuckDBPyConnection API, and require all the users to just use `PathLike::Create(object, *this)` directly.

Reply:
I think this is more convenient than requiring users to take extra steps.
It doesn't require the API to change, but now accepts more options